### PR TITLE
ref(*): pass io.ReadCloser instead of string body

### PR DESCRIPTION
**Pull request by Joshua-Anderson**
Requested by @arschles 

Note that this PR removes the current system of error messages, because it is incompatible with the `io.ReadCloser` refactor.
My work over the next few days will be to implement a new error system based on error constants that can be checked by sdk users and that will be compatible with `io.Readers`. If we don't want to temporarily remove verbose error messages, I can get this reviewed, then submit a new PR with both this refactor and the error messages refactor.

Also note that this does not affect `LimitedRequest()`. This is because it needs to parse the json body in order to work, meaning that the response already has to be converted to a string. I'm open for ways that this could be better implemented.

I'm returning the entire `*http.Response` object as opposed to just the body, because some of the information, such as response code, can be used to return specific error messages for an action with my error messages rewrite.

### CODE REVIEW

File: apps/apps.go [hunk 1]
```diff
@@ -103,15 +113,23 @@ func Logs(c *deis.Client, appID string, lines int) (string, error) {
...
-	body, err := c.BasicRequest("GET", u, nil)
+	res, err := c.Request("GET", u, nil)
+	if err != nil {
+		return "", ErrNoLogs
+	}
+	// Fix json.Decoder bug in <go1.7
+	defer func() {
+		io.Copy(ioutil.Discard, res.Body)
+		res.Body.Close()
+	}()
```
Comment (arschles) [inline apps/apps.go]: don't you want to schedule this after the `ioutil.ReadAll` returns?

File: apps/apps.go [hunk 2]
```diff
@@ -103,15 +113,19 @@ func Logs(c *deis.Client, appID string, lines int) (string, error) {
...
 
+	body, err := ioutil.ReadAll(res.Body)
 	if err != nil || len(body) < 1 {
-
 		return "", ErrNoLogs
 	}
 
 	// We need to trim a few characters off the front and end of the string
-	return body[2 : len(body)-1], nil
+	return string(body[2 : len(body)-1]), nil
```
Comment (arschles) [inline apps/apps.go]: can you do a bounds check here?

File: domains/domains.go [hunk 3]
```diff
@@ -37,23 +39,30 @@ func New(c *deis.Client, appID string, domain string) (api.Domain, error) {
...
-
+	res, err := c.Request("POST", u, body)
 	if err != nil {
 		return api.Domain{}, err
 	}
+	// Fix json.Decoder bug in <go1.7
+	defer func() {
+		io.Copy(ioutil.Discard, res.Body)
+		res.Body.Close()
+	}()
```
Comment (arschles) [inline domains/domains.go]: shouldn't this gouroutine be scheduled after the json decode line?

File: keys/keys.go [hunk 4]
```diff
@@ -29,14 +31,18 @@ func New(c *deis.Client, id string, pubKey string) (api.Key, error) {
...
-
+	res, err := c.Request("POST", "/v2/keys/", body)
 	if err != nil {
 		return api.Key{}, err
 	}
+	// Fix json.Decoder bug in <go1.7
+	defer func() {
+		io.Copy(ioutil.Discard, res.Body)
+		res.Body.Close()
+	}()
```
Comment (arschles) [inline keys/keys.go]: same here, should this be scheduled after the decode step?

File: apps/apps.go (hunk 1 above)
Comment (arschles) [inline apps/apps.go]: sorry, I misread and assumed the `defer` was actually a `go`. ignore me

File: domains/domains.go (hunk 3 above)
Comment (arschles) [inline domains/domains.go]: sorry, I misread and assumed the `defer` was actually a `go`. ignore me

File: keys/keys.go (hunk 4 above)
Comment (arschles) [inline keys/keys.go]: sorry, I misread and assumed the `defer` was actually a `go`. ignore me

**Comment by arschles** [comment]
code LGTM after https://github.com/deis/controller-sdk-go/pull/11#discussion_r67722843 is addressed

File: apps/apps.go (hunk 2 above)
Comment (Joshua-Anderson) [inline apps/apps.go]: I'm now doing a bounds check in the previous if statement.

**Pull request merged**